lisa/transformers/kernel_installer.py

```diff
@@ -0,0 +1,308 @@
+    source: str = field(
+        default="proposed",
+        metadata=schema.metadata(
+            required=True,
+        ),
+    )
+    is_proposed: bool = True
+    delay_parsed: CatchAll = field(default_factory=dict)  # type: ignore
+
+
+@dataclass_json()
+@dataclass
+class PpaInstallerSchema(RepoInstallerSchema):
+    openpgp_key: str = ""
+    ppa_url: str = field(
+        default="",
+        metadata=schema.metadata(
+            required=True,
+        ),
+    )
+    is_proposed: bool = False
+
+    delay_parsed: CatchAll = field(default_factory=dict)  # type: ignore
+
+    def __post_init__(self, *args: Any, **kwargs: Any) -> None:
+        add_secret(self.ppa_url, PATTERN_HEADTAIL)
+
+
+@dataclass_json
+@dataclass
+class KernelInstallerTransformerSchema(schema.Transformer):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=schema.metadata(required=True)
+    )
+    installer: Optional[BaseInstallerSchema] = field(
+        default=None, metadata=schema.metadata(required=True)
+    )
+
+
+class BaseInstaller(subclasses.BaseClassWithRunbookMixin):
+    def __init__(
+        self,
+        runbook: Any,
+        node: RemoteNode,
+        parent_log: Logger,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(runbook, *args, **kwargs)
+        self._node = node
+        self._log = get_logger("kernel_installer", parent=parent_log)
+
+    def validate(self) -> None:
+        raise NotImplementedError()
+
+    def install(self) -> None:
+        raise NotImplementedError()
+
+
+class KernelInstallerTransformer(Transformer):
+    @classmethod
+    def type_name(cls) -> str:
+        return "kernel_installer"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return KernelInstallerTransformerSchema
+
+    @property
+    def _output_names(self) -> List[str]:
+        return []
+
+    def _internal_run(self) -> Dict[str, Any]:
+        runbook: KernelInstallerTransformerSchema = self.runbook
+        assert runbook.connection, "connection must be defined."
+        assert runbook.installer, "installer must be defined."
+
+        node: RemoteNode = cast(
+            RemoteNode, Node.create(-1, runbook.connection, logger_name="kernel_node")
+        )
+        node.set_connection_info_by_runbook()
+        node.initialize()
+
+        uname = node.tools[Uname]
+        self._log.info(
+            f"kernel version before install: {uname.get_linux_information()}"
+        )
+        factory = subclasses.Factory[BaseInstaller](BaseInstaller)
+        installer = factory.create_by_runbook(
+            runbook=runbook.installer, node=node, parent_log=self._log
+        )
+
+        installer.validate()
+        installer.install()
+
+        self._log.info("rebooting")
+        node.reboot()
+        self._log.info(
+            f"kernel version after install: "
+            f"{uname.get_linux_information(force_run=True)}"
+        )
```

Comment:
There is log to output the before/after installation. Considering sometime may be the version is latest already, so there is no assertion to check if the version is changed or not. If it's really needed, let me know, I can add a validation, and it can be skipped.